src/collider/src/collider.cpp

```diff
@@ -188,15 +190,16 @@ struct FindCollisions {
     int top = -1;
     // Depth-first search
     int node = kRoot;
+    const int queryIdx = thrust::get<1>(query);
+    // same implies that this query do not have any collision
+    if (!preallocate && counts[queryIdx] == counts[queryIdx + 1]) return;
     while (1) {
       int internal = Node2Internal(node);
       int child1 = internalChildren_[internal].first;
       int child2 = internalChildren_[internal].second;
 
       int traverse1 = RecordCollision(child1, query);
-      if (traverse1 < 0) return;
       int traverse2 = RecordCollision(child2, query);
-      if (traverse2 < 0) return;
```

Comment:
Why is this no longer useful?

Reply:
because `RecordCollision` can no longer do early exit due to insufficient memory, it will calculate the number of elements and allocate exactly that

Reply:
Ah, right; I'd forgotten about that. Thanks!